blockchain/validate.go

```diff
@@ -774,14 +774,34 @@ func (b *BlockChain) checkBlockContext(block *btcutil.Block, prevNode *blockNode
 
 	fastAdd := flags&BFFastAdd == BFFastAdd
 	if !fastAdd {
+		medianTime := b.BestSnapshot().MedianTime
+
+		// Obtain the latest state of the deployed CSV soft-fork in
+		// order to properly guard the new validation behavior based on
+		// the current BIP 9 version bits state.
+		csvState, err := b.ThresholdState(chaincfg.DeploymentCSV)
```

Comment:
I believe this will end up dead locking because the chain state lock is already held and the exported `ThresholdState` takes the lock.  Given the usefulness of being able to check the state based on a deployment as the exported version does, I suspect we might need to refactor it slightly such that there is a `thresholdHoldForDeployment` or something similar.